### PR TITLE
[Jetpack plugin install prompt] Show prompt when Jetpack is not connected

### PR DESCRIPTION
**Pull request by staskus**
This is part of [Show Jetpack plugin install prompt during Jetpack app login process](https://github.com/wordpress-mobile/WordPress-iOS/issues/19213). Merging to the parent branch.

## Description

Show Jetpack plugin install prompt when Jetpack is not connected. Use the same presentation logic throughout all the Jetpack plugin install prompts: 1) Log in 2) Stats 3) Notifications

There're 3 states of Jetpack plugin that I identified:
1. `blog.jetpack.installed` true if Jetpack plugin is installed and active
2. `blog.jetpack.connected` true if "Set up" of Jetpack is done (completing log in to WP.com account is not necessary for it to become true)
3. `blog.account` has value if WP.com account is connected

Previously, [the logic to present Jetpack plugin install prompt used](https://github.com/wordpress-mobile/WordPress-iOS/pull/18319/commits/746d706fd8189db8f26e3e00f7601244334a32a9) `jetpack.installed` check. However, it creates differences between the login prompt and other prompts. So the user might not be prompted to install after logging in but would be prompted to install after trying to access stats. Therefore, using `jetpack.connected` check instead. 

### Case 1: Jetpack plugin not installed
1. Fresh install Jetpack App
2. Log into a self-hosted site that doesn't have the Jetpack plugin installed
3. Install prompt should appear after login

### Case 2: Jetpack plugin is installed and not connected
1. Fresh install Jetpack App
2. Log into a self-hosted site that has Jetpack installed and active but not connected
3. Install prompt should appear after login

### Case 3: Jetpack plugin is installed and connected
1. Fresh install Jetpack App
2. Log into a self-hosted site that has Jetpack installed and connected (doing "Set Up" step without completing WP.com log in)
3. Install prompt should not appear after login

## Regression Notes

1. Potential unintended areas of impact

None

2. What I did to test those areas of impact (or what existing automated tests I relied on)

None

3. What automated tests I added (or what prevented me from doing so)

Expanded `JetpackInstallPromptTests` tests

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Images

1. Jetpack installed, Jetpack not connected
Jetpack mobile + Set Up Jetpack on web

If we click "Set Up" / "Install" but don't successfully log in into account we'll arrive at such state:
image

### CODE REVIEW

**Review by guarani — COMMENTED** [review]
👋 @staskus, I'll have to leave this till tomorrow (Thu) to review, but I gave it a glance and it's looking good!

**Review by guarani — COMMENTED** [review]
> ### Case 3: Jetpack plugin is installed and connected
> 1. Fresh install Jetpack App
> 2. Log into a self-hosted site that has Jetpack installed and connected (doing "Set Up" step without completing WP.com log in)
> 3. Install prompt should not appear after login

In this case, the user has installed Jetpack and even connected it to WordPress.com. However, they haven't connected it to **their** WordPress.com account. This means they still haven't gotten the most important of Jetpack's features: login via WP.com credentials, Stats, Notifications, etc.
Should we show a prompt in this scenario (e.g. a  "Jetpack plugin WP.com account connection" prompt), to encourage the user to finish the connection and unlock all of these Jetpack features?

File: WordPress/Classes/ViewRelated/Jetpack/JetpackInstallPromptSettings.swift [hunk 1]
```diff
@@ -31,7 +31,7 @@ public final class JetpackInstallPromptSettings {
             return false
         }
 
-        return !jetpack.isInstalled && blog.isAdmin && !promptWasDismissed(for: blog)
+        return !jetpack.isConnected && blog.isAdmin && !promptWasDismissed(for: blog)
```
Comment (guarani) [inline WordPress/Classes/ViewRelated/Jetpack/JetpackInstallPromptSettings.swift:34]: To clarify my above comment, the change would look something like this:
```suggestion
        let isJetpackConnectedToWPAccount = jetpack.isInstalled && blog.account != nil
        return !isJetpackConnectedToWPAccount && blog.isAdmin && !promptWasDismissed(for: blog)
```
I'm aware that this is the "Jetpack **_Install_** Prompt", so having it also prompt users to connect Jetpack to their WP.com accounts could be scope creep. 
That said, a user who doesn't connect their WP.com account will need to log in via their site credentials the next time they want to log in to their site via the app.

Comment (staskus) [reply WordPress/Classes/ViewRelated/Jetpack/JetpackInstallPromptSettings.swift:34]: Thanks for the review and suggestions!

> Should we show a prompt in this scenario (e.g. a "Jetpack plugin WP.com account connection" prompt), to encourage the user to finish the connection and unlock all of these Jetpack features?

Yes, we definitely should. It will be implemented in the next phase. 

> I'm aware that this is the "Jetpack Install Prompt", so having it also prompt users to connect Jetpack to their WP.com accounts could be scope creep.

Yes, that is an increase in scope.  To implement such logic properly, we would need to:
- Add  presentation logic to show WP.com login screen instead of Install screen,  if Jetpack is connected but the WP.com account doesn't exist
- If we want to have friendly UX, replace "Install prompt" with some new "Login to WP.com" prompt, if Jetpack is connected but the WP.com account doesn't exist

If we change the code as in the suggestion:
- "Install prompt" is still shown in the case if Jetpack is connected and the WP.com account doesn't exist
- When we click "Install" we are taken to the installation flow
- Installation flow fails, since Jetpack is already installed and connected

drawing

Therefore, such change wouldn't be enough for now.

**Review by guarani — APPROVED** [review]
Tested and everything is working great!

- [x] Case 1: Jetpack plugin not installed
- [x] Case 2: Jetpack plugin is installed and not connected
- [x] Case 3: Jetpack plugin is installed and connected
